### ISSUE TITLE
keep getting a TypeError exception when running airline sample app.

### ISSUE THREAD

**Issue by AliTajeldin** [TresAmigosSD/SMV#702]
The exception is being ignored but is rather annoying.
```
Exception TypeError: 'isinstance() arg 2 must be a class, type, or tuple of classes and types' in <function <lambda> at 0x10bb2a6e0> ignored
Exception TypeError: 'isinstance() arg 2 must be a class, type, or tuple of classes and types' in <function <lambda> at 0x10bb2a758> ignored
Exception TypeError: 'isinstance() arg 2 must be a class, type, or tuple of classes and types' in <function <lambda> at 0x10bb2a5f0> ignored
Exception TypeError: 'isinstance() arg 2 must be a class, type, or tuple of classes and types' in <function <lambda> at 0x10bb2a7d0> ignored

```

**Comment by laneb** [comment]
What's the airline sample app? I don't see it in the templates.

**Comment by AliTajeldin** [comment]
Seems to be a py4j issue: https://github.com/bartdag/py4j/issues/106

**Comment by Kai-Chen** [comment]
> Interestingly, adding an extra, unused py4j import in test.py fixes the exception and causes all of the py4j modules to be cleaned up after the test module and JavaGateway instance.

Reading that quote gives me an odd urge to laugh out loud :)
Worth a try here?

It should be fixed in spark 2.1, which bundles py4j 0.10.4.

**Comment by AliTajeldin** [comment]
closing as will not fix.
